src/sagas/pushNotification.js

```diff
@@ -0,0 +1,406 @@
+ *   titleLocKey: 'new_transaction_received_title',
+ *   txId: 'txId1',
+ * }
+ * @inner
+ */
+const messageHandler = async (message) => {
+  const { data } = message;
+  if (!localization.hasKey(data.titleLocKey)) {
+    console.debug('unknown message titleLocKey', data.titleLocKey);
+    return;
+  }
+
+  if (!localization.hasKey(data.bodyLocKey)) {
+    console.debug('unknown message bodyLocKey', data.bodyLocKey);
+    return;
+  }
+
+  const bodyArgs = JSON.parse(data.bodyLocArgs);
+  const title = localization.getMessage(data.titleLocKey);
+  const body = localization.getMessage(data.bodyLocKey, bodyArgs);
+
+  notifee.displayNotification({
+    title,
+    body,
+    android: {
+      channelId: TRANSACTION_CHANNEL_ID,
+    },
+  });
+};
+
+/**
+ * @returns {Promise<boolean>} true if the device is registered on the FCM, false otherwise
+ */
+const confirmDeviceRegistrationOnFirebase = async () => {
+  try {
+    // Make sure deviceId is registered on the FCM
+    if (messaging().isDeviceRegisteredForRemoteMessages) {
+      await messaging().registerDeviceForRemoteMessages();
+    }
+    return true;
+  } catch (error) {
+    console.error(`Error confirming the device is registered on FCM: ${error.message}`, error);
+    return false;
+  }
+};
+
+const installForegroundListener = () => {
+  try {
+    // Add listeners for push notifications on foreground and background
+    messaging().onMessage(onForegroundMessage);
+  } catch (error) {
+    console.error(`Error installing foreground listener to push notification: ${error.message}`, error);
+  }
+};
+
+/**
+ * @returns {Promise<boolean>} true if the channel was created or already exists, false otherwise
+ */
+const createChannelIfNotExists = async () => {
+  try {
+    const hasTransactionChannel = await notifee.isChannelCreated(TRANSACTION_CHANNEL_ID);
+    if (!hasTransactionChannel) {
+      await notifee.createChannel({
+        id: TRANSACTION_CHANNEL_ID,
+        name: TRANSACTION_CHANNEL_NAME,
+      });
+    }
+    return true;
+  } catch (error) {
+    console.error(`Error creating channel for push notification: ${error.message}`, error);
+    return false;
+  }
+};
+
+/**
+ * This function is called when the wallet is initialized with success.
+ */
+export function* onAppInitialization() {
+  yield take(types.START_WALLET_SUCCESS);
+
+  // If the channel is not created, we should not continue.
+  const isChannelCreated = yield call(createChannelIfNotExists);
+  if (!isChannelCreated) {
+    console.debug('Halting push notification initialization because the channel was not created.');
+    return;
+  }
+
+  // If the device is not registered on FCM, we should not continue.
+  const isDeviceRegistered = yield call(confirmDeviceRegistrationOnFirebase);
+  if (!isDeviceRegistered) {
+    console.debug('Halting push notification initialization because the device is not registered on FCM.');
+    return;
+  }
+
+  yield call(installForegroundListener);
```

Comment:
Should we stop the push notification process here as well if it fails?

Reply:
I have implemented the error capture here in the PR for the tx details:
https://github.com/HathorNetwork/hathor-wallet-mobile/pull/233/files#diff-b2f41e02468f99f5c9c39db9c11e7c79fbde4091cf522468d84af47845be7adaR155